src/store.cc

```diff
@@ -1714,15 +1714,14 @@ StoreEntry::startWriting()
     rep->packHeadersUsingSlowPacker(*this);
     mem_obj->markEndOfReplyHeaders();
 
+    // Same-worker collapsing risks end with the receipt of the headers.
+    // SMP collapsing risks remain until the headers are actually cached, but
+    // that event is announced via CF-agnostic disk I/O broadcasts.
```

Comment:
This implies that there is some code that clears ENTRY_REQUIRES_COLLAPSING in `anchor.basics`, but I could not find it. Perhaps we do not need it anymore? Also what about SMP shm-only configurations?

Reply:
> This implies that there is some code that clears ENTRY_REQUIRES_COLLAPSING in `anchor.basics`

I do not see such an implication. To me, the comment just explains why we are clearing collapsing risks for this worker (i.e. this StoreEntry object) but not notifying other workers yet. Please detail how you get from this comment to anchor.basics clearing, and I will try to polish the comment to remove that implication (that I do not yet see).

AFAICT, there should be no code that explicitly updates ENTRY_REQUIRES_COLLAPSING in anchor.basics because we are not supposed to use anchor.basics to signal these requirements -- that is one of the simplifications/improvements contained in this PR.


> Perhaps we do not need it anymore?

I think this `if` statement is still needed to end same-worker collapsing risks. Or did you mean something else by `it`?

> Also what about SMP shm-only configurations?

I would expect MemStore to broadcast any changes as the headers get cached there. The comment should not be disk-specific though. I suggest `s/disk I.O/Store writing/` to fix that.